src/stacked_borrows.rs

```diff
@@ -366,7 +399,12 @@ impl<'tcx> Stack {
             let first_incompatible_idx = self.find_first_write_incompatible(granting_idx);
             for item in self.borrows.drain(first_incompatible_idx..).rev() {
                 trace!("access: popping item {:?}", item);
-                Stack::check_protector(&item, Some((tag, access)), global)?;
+                Stack::check_protector(
+                    &item,
+                    Some((tag, alloc_id, alloc_range, offset, access)),
+                    global,
+                )?;
+                global.log_invalidation(item.tag, alloc_id, alloc_range, threads);
```

Comment:
Can we call this from within `check_protector`? all the information should be available (beyond the span)

Reply:
We _could_, but I don't think checking protectors is the operation that invalidates tags so to my eyes that's a confusing way to factor this.